src/lib/snarky_bowe_gabizon_hash/snarky_bowe_gabizon_hash.ml

```diff
@@ -0,0 +1,243 @@
+open Core_kernel
+open Fold_lib
+open Snarky
+
+(* TODO: Have a compatibility layer so we only need the module M. *)
+module Make
+    (M : Snark_intf.Run with type prover_state = unit)
+    (Impl : Snark_intf.S with type field = M.field)
```

Comment:
It looks like you could get rid of this and use `M.Internal_basic` if you tweak the signature of `Snarky_blake2.Make` to take a `Snarm_intf.Basic` instead of a `Snarky_intf.S`.